packages/vertexai/src/models/vertexai-model.ts

```diff
@@ -68,10 +68,16 @@ export abstract class VertexAIModel {
         VertexAIErrorCode.NO_PROJECT_ID,
         `The "projectId" field is empty in the local Firebase config. Firebase VertexAI requires this field to contain a valid project ID.`
       );
+    } else if (!vertexAI.app?.options?.appId) {
+      throw new VertexAIError(
```

Comment:
In light of Paul's comment about `automaticDataCollectionEnabled`, not sure if we throw this error if that's false, since we don't actually need appId in that case? Kind of weird edge case though.

Reply:
The `firebaseAppConfig` obtained from Firebase Console and Terraform always contains an `appId`, so unless the developer deliberately removes the `appId` (not sure why since it's not sensitive), it should always be present. AppId is not just for telemetry purposes. Thus, stronger validation is better here.